heap/heap.py

```diff
@@ -1,21 +1,49 @@
 class Heap:
-  def __init__(self):
-    self.storage = []
+    def __init__(self):
+        self.storage = []
 
-  def insert(self, value):
-    pass
+    def insert(self, value):
+        self.storage.append(value)
+        self._bubble_up(len(self.storage)-1)
 
-  def delete(self):
-    pass
+    def delete(self):
+        # remove the first item in array and place the last item in the first place
+        self.storage[0] = self.storage[len(self.storage)-1]
+        self.storage.pop()
 
-  def get_max(self):
-    pass
+        index = 0
+        # loop with _sift_down
 
-  def get_size(self):
-    pass
+        while index < len(self.storage) - 1:
+            index = self._sift_down(index)
 
-  def _bubble_up(self, index):
-    pass
+    def get_max(self):
+        if self.storage[0]:
+            return self.storage[0]
+        else:
+            return None
 
-  def _sift_down(self, index):
-    pass
+    def get_size(self):
+        return len(self.storage)
+
+    def _bubble_up(self, index):
+
+        while self.storage[index] > self.storage[(index-1)//2]:
```

Comment:
Some comments would be nice here to explain what is actually being done (i.e. the mechanism of bubble_up).

Reply:
comments are added :)

Reply:
sweet!